### PR TITLE
Idiomatic errors

### PR DESCRIPTION
**Pull request by toqueteos**
Hello again!

Here's an unfinished PR for moving all errors into `ErrXXX` values and all the `errors.New` using string interpolation to `fmt.Errorf`.

I'll try my best to have it finished this evening (that should be around 20:00 GMT+2)

### CODE REVIEW

File: connect_token_private.go [hunk 1]
```diff
@@ -83,18 +82,18 @@ func (p *ConnectTokenPrivate) Write() ([]byte, error) {
 }
 
 // Encrypts, in place, the TokenData buffer, assumes Write() has already been called.
-func (token *ConnectTokenPrivate) Encrypt(protocolId, expireTimestamp, sequence uint64, privateKey []byte) error {
+func (p *ConnectTokenPrivate) Encrypt(protocolId, expireTimestamp, sequence uint64, privateKey []byte) error {
```
Comment (toqueteos) [inline connect_token_private.go:85]: Not related to errors but I changed this so all methods would be consistent

**Comment by toqueteos** [comment]
Sorry for delay. It's done! @wirepair

Now  that all possible exposable errors are exposed as package level constants we can now use those in tests too! I'm sure users of the library will be happy of having them.

**ASIDE:** Repository has 75% coverage which is pretty good!

**Review by wirepair — APPROVED** [review]

**Comment by wirepair** [comment]
Thanks for the changes, they look great!

**Comment by wirepair** [comment]
Yeah i've been meaning to add more tests, there's quite a few things i'd like to change about the design as well to clean it up a bit I just haven't had the time yet.

**Pull request merged**